File: java/src/java/adapter/Adapter.java
package adapter;

public class Adapter implements TriplePin{

    private final DualPin dualPinDevice;

    public Adapter(DualPin dualPinDevice) {
        this.dualPinDevice = dualPinDevice;
    }

    @Override
    public void electrify(int l, int n, int e) {
        dualPinDevice.electrify(l, n);
    }
}

File: golang/adapter/adapter.go
package adapter

import "fmt"

type TriplePin interface { // 三线插口接口
	Electrify(l int, n int, e int) // l 火线 ， n 零线， e 地线
}

type DualPin interface { // 两线插口接口
	Electrify(l int, n int) // l 火线 ， n 零线
}

type TV struct {
}

func (t *TV) Electrify(l int, n int) {
	fmt.Printf("火线通电：%d\n", l)
	fmt.Printf("零线通电：%d\n", n)
}

type Adapter struct {
	dualPinDevice DualPin
}

func (a *Adapter) Electrify(l int, n int, e int) {
	a.dualPinDevice.Electrify(l, n)
}
